### PR TITLE
WT-7834 Colorize build output with CMake

### PR DESCRIPTION
**Pull request by mariammojid**
In CMakeLists file, enabled diagnostics coloring for clang and gcc when compilation fails.

Here's what it should look like:
Screenshot 2023-12-13 at 1 50 53 pm

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -83,6 +83,18 @@ else()
     set(CMAKE_CXX_EXTENSIONS OFF)
 endif()
 
+# Colorize the CMake error output
```
Comment (ruby-oujo) [inline CMakeLists.txt:86]: perhaps change CMake to build

Comment (mariammojid) [reply CMakeLists.txt]: done, thanks

File: CMakeLists.txt [hunk 2]
```diff
@@ -83,6 +83,18 @@ else()
...
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fdiagnostics-color=always" CACHE STRING "" FORCE)
+elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fcolor-diagnostics" CACHE STRING "" FORCE)
+endif()
+if("${CMAKE_C_COMPILER_ID}" STREQUAL "GNU")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fdiagnostics-color=always" CACHE STRING "" FORCE)
+elseif("${CMAKE_C_COMPILER_ID}" STREQUAL "Clang")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fcolor-diagnostics" CACHE STRING "" FORCE)
+endif()
+
```
Comment (ruby-oujo) [inline CMakeLists.txt:97]: We can try simplifying this to something like:

```
if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_C_COMPILER_ID}" STREQUAL "GNU")
    add_compile_options(-fdiagnostics-color=always)
elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_C_COMPILER_ID}" STREQUAL "Clang")
    add_compile_options(-fcolor-diagnostics)
endif()

```

**Review by ajmorton — APPROVED** [review]
LGTM! 🎉

File: CMakeLists.txt (hunk 2 above)
Comment (mariammojid) [reply CMakeLists.txt:92]: thank you for the suggestion!

**Review by ruby-oujo — APPROVED** [review]
LGTM! Amazing work woo!

**Comment by mariammojid** [comment]
evergreen merge

**Pull request merged**